File: typescript-playwright/src/helpers/UiAuthHelper.ts
import type { Page } from "@playwright/test";
import type { AuthHelper } from "./AuthHelper";
import { step } from "../utils/decorators";

export class UiAuthHelper {
  private readonly page: Page;
  private readonly authHelper: AuthHelper;

  constructor(
    page: Page,
    authHelper: AuthHelper,
  ) {
    this.page = page;
    this.authHelper = authHelper;
  }

  @step("Авторизоваться как админ")
  async loginAsAdmin(redirectUrl = "/admin-panel/upload"): Promise<void> {
    const token = await this.authHelper.getAdminToken();

    await this.page.goto("/");
    await this.page.evaluate((t) => {
      localStorage.setItem("accessToken", t);
    }, token);

    await this.page.goto(redirectUrl);
    await this.page.waitForLoadState("load");
  }
}

File: python/src/helpers/ui_auth_helper.py
import allure
from playwright.async_api import Page

from src.helpers.auth_helper import AuthHelper

ROOT_PATH: str = "/"
DEFAULT_ADMIN_REDIRECT: str = "/admin-panel/upload"


class UiAuthHelper:
    def __init__(self, page: Page, auth_helper: AuthHelper):
        self.page = page
        self.auth_helper = auth_helper

    @allure.step("Авторизоваться как админ")
    async def login_as_admin(self, redirect_url: str = DEFAULT_ADMIN_REDIRECT) -> None:
        token = self.auth_helper.get_admin_token()

        await self.page.goto(ROOT_PATH)

        await self.page.evaluate("token => localStorage.setItem('accessToken', token)", token)

        await self.page.goto(redirect_url)
        await self.page.wait_for_load_state("load")
